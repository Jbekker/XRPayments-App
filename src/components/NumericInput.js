import React, { PureComponent } from "react";
import { StyleSheet, View, Text, TouchableOpacity, Animated, DeviceEventEmitter, Image } from "react-native";

import { AppSizes } from "@theme";

import { ScaleFont, Px2Dp, Format } from "@libs/utils";

const styles = StyleSheet.create({
    view: {
        backgroundColor: "#FFF",
        justifyContent: "center",
        width: AppSizes.screen.width * 0.5,
        height: AppSizes.screen.height * 0.1,
        textAlign: "center",
        borderWidth: 1,
        borderColor: "#c6c6c6",
        borderRadius: AppSizes.screen.width * 0.7 * AppSizes.screen.height * 0.1,
    },
    textInputWrap: {
        height: AppSizes.screen.height * 0.1,
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        paddingLeft: Px2Dp(10),
    },
    cursorWrap: {
        height: AppSizes.screen.height * 0.1,
        flexDirection: "row",
        alignItems: "center",
    },
    cursor: {
        color: "#000",
        fontSize: ScaleFont(35),
        fontWeight: "300",
    },
    placeholder: {
        color: "#333",
        fontSize: ScaleFont(20),
        fontWeight: "700",
        position: "absolute",
        textAlign: "center",
        left: ScaleFont(25),
    },
    currency: {
        position: "absolute",
        tintColor: "#000",
        right: AppSizes.screen.width * 0.02,
        width: AppSizes.screen.width * 0.031,
        height: AppSizes.screen.width * 0.031,
    },
    value: {
        fontWeight: "700",
        fontSize: ScaleFont(35),
        color: "#21B466",
    },
});

class NumericInput extends PureComponent {
    constructor(props) {
        super(props);

        this.state = {
            fadeAnim: new Animated.Value(0),
            valueArr: props.value || [],
        };
    }
    componentDidMount() {
        this.inputEvent();

        this.animation();
    }

    static getDerivedStateFromProps(props, state) {
        if (props.value.split("") !== state.valueArr) {
            return {
                valueArr: props.value.split(""),
            };
        }

        return null;
    }

    inputEvent() {
        const self = this;
        self.subscription = DeviceEventEmitter.addListener("numericKeyboardInput", data => {
            self.setState({
                valueArr: data,
            });
        });
    }

    animation() {
        let that = this;
        this.animated = Animated.loop(
            Animated.sequence([
                Animated.timing(that.state.fadeAnim, {
                    toValue: 1,
                    duration: 600,
                    seNativeDriver: true,
                }),
                Animated.timing(that.state.fadeAnim, {
                    toValue: 0,
                    duration: 600,
                    seNativeDriver: true,
                }),
            ]),
            {
                iterations: 400,
            },
        ).start();
    }

    componentWillUnmount() {
        this.subscription.remove();
    }

    renderValue() {
        const { valueArr } = this.state;

        if (valueArr.length > 0) {
            return <Text style={[styles.value, this.props.valueStyle]}>{Format(valueArr.join(""))}</Text>;
        } else {
            return null;
        }
    }

    render() {
        return (
            <View style={[styles.view]}>
                <TouchableOpacity style={styles.textInputWrap}>
                    {this.renderValue()}

                    <Text style={[styles.placeholder, this.props.valueStyle || {}]}>{this.props.placeholder}</Text>

                    <Image
                        style={[styles.currency]}
                        resizeMode={"contain"}
                        source={require("../assets/images/icon-pay.png")}
                    />

                    {!this.props.cursorLock && !this.props.caretHidden ? (
                        <Animated.View style={[styles.cursorWrap, { opacity: this.state.fadeAnim }]}>
                            <Text style={styles.cursor}>|</Text>
                        </Animated.View>
                    ) : null}
                </TouchableOpacity>
            </View>
        );
    }
}

export default NumericInput;
