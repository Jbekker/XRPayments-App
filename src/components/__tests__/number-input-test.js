import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import NumericInput from "@components/NumericInput";

it("renders correctly", () => {
    const tree = renderer.create(<NumericInput />).toJSON();
    expect(tree).toMatchSnapshot();
});
