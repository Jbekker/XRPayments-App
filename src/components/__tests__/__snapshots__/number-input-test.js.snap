// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFF",
        "borderColor": "#c6c6c6",
        "borderRadius": 70035,
        "borderWidth": 1,
        "color": "#5a5a5a",
        "height": 133.4,
        "justifyContent": "center",
        "textAlign": "center",
        "width": 375,
      },
    ]
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 40,
        "justifyContent": "center",
        "opacity": 1,
        "paddingLeft": 10,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#333",
            "fontSize": 15,
            "fontWeight": "700",
            "left": 18.75,
            "position": "absolute",
            "textAlign": "center",
          },
          Object {},
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "color": "#333",
            "fontSize": 18.75,
            "position": "absolute",
            "right": 25,
            "textAlign": "center",
          },
        ]
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 40,
          "opacity": 0,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3E77E6",
            "fontSize": 26.25,
            "fontWeight": "300",
          }
        }
      >
        |
      </Text>
    </View>
  </View>
</View>
`;
